avocado/core/test.py

```diff
@@ -782,7 +782,11 @@ def skip(self, message=None):
         :param message: an optional message that will be recorded in the logs
         :type message: str
         """
-        raise exceptions.TestSetupSkip(message)
+        dep_msg = '[WARNING: self.skip() will be deprecated, please ' \
+                  'use self.cancel()] '
```

Comment:
One more thing, you should probably change the `examples/tests/skiponsetup.py` in the same commit to demonstrate the required change.